libqtile/group.py

```diff
@@ -519,6 +519,30 @@ def cmd_info_by_name(self, name):
             if win.name == name:
                 return win.info()
 
+    def cmd_focus_by_index(self, index: int) -> None:
+        """
+        Change to the window at the specified index in the current group.
+        """
+        windows = self.windows
+        if index < 0 or index > len(windows) - 1:
+            return
+
+        self.focus(windows[index])
+
+    def cmd_swap_window_order(self, new_location: int) -> None:
+        """
+        Change the order of the current window within the current group.
+        """
+        if new_location < 0 or new_location > len(self.windows) - 1:
+            return
+
+        windows = self.windows
+        current_window_index = windows.index(self.current_window)
+
+        temp = windows[current_window_index]
+        windows[current_window_index] = windows[new_location]
+        windows[new_location] = temp
```

Comment:
You can avoid the temp:
```suggestion
windows[current_window_index], windows[new_location] = windows[new_location], windows[current_window_index]
```